File: android/gemcore/src/main/kotlin/com/gemwallet/android/ext/GemNameService.kt
package com.gemwallet.android.ext

import com.wallet.core.primitives.Chain
import uniffi.gemstone.GemNameRecordState
import uniffi.gemstone.GemNameServiceInterface
import uniffi.gemstone.GemRecipientValidation

fun GemNameServiceInterface.validateRecipient(chain: Chain, input: String, state: GemNameRecordState): GemRecipientValidation =
    validateRecipient(chain.string, input, state)

suspend fun GemNameServiceInterface.getNameRecord(name: String, chain: Chain): GemNameRecordState =
    getNameRecord(name, chain.string)

File: ios/Packages/GemstonePrimitives/Sources/Services/GemNameService.swift
// Copyright (c). Gem Wallet. All rights reserved.

import protocol Gemstone.GemNameServiceProtocol
import Primitives

public extension GemNameServiceProtocol {
    func getNameRecord(name: String, chain: Chain) async throws -> NameRecord? {
        try await getNameRecord(name: name, chain: chain.rawValue).map { $0.map() }
    }
}
